### ISSUE TITLE
Fetched nested objects aren't fetched after save

### ISSUE THREAD

**Issue by dpoetzsch** [HustleInc/parse-mockdb#44]
When saving an object with nested objects that are already fetched, exactly those nested objects should still be fetched after the save. Check out the following unit test that asserts this:

```javascript
it('should keep fetched nested objects fetched after the save', function() {
  return new Item().save({
    price: 45
  }).then(function(item0) {
    return new Item().save({
      price: 50
    }).then(function(item2) {
      return new Item({
        price: 55
      }).save().then(function(item3) {
        const brand = new Brand();
        const item1 = new Item();
        item1.id = item0.id; // create pointer to item0
        brand.set("items", [item1, item2, item3]);
        return brand.save();
      });
    });
  }).then(function(brand) {
    assert.equal(brand.get("items")[0].get("price"), undefined);
    assert.equal(brand.get("items")[1].get("price"), 50);
    assert.equal(brand.get("items")[2].get("price"), 55);

    brand.get("items")[0].set("price", 30);
    brand.set("name", "foo");
    return brand.save();
  }).then(function(sbrand) {
    assert.equal(sbrand.get("items")[0].get("price"), undefined);
    assert.equal(sbrand.get("items")[1].get("price"), 50);
    assert.equal(sbrand.get("items")[2].get("price"), 30);
  });
});
```

**Comment by TylerBrock** [comment]
Does it work that way on api.parse.com or does it also decay to a pointer?

**Comment by dpoetzsch** [comment]
Yes, I tested the behavior with parse hosted cloud code.

**Comment by TylerBrock** [comment]
This should be fixed as of the merging of #50. I'll do a release to incorporate the change.
